src/main/groovy/com/google/protobuf/gradle/ToolsLocator.groovy

```diff
@@ -108,4 +108,13 @@ class ToolsLocator {
       }
     }
   }
+
+  static List<String> artifactParts(String artifactCoordinate) {
+    final validName = '[+-.0-9A-Z_a-z]+'
+
+    ((artifactCoordinate =~ /(${validName}):(${validName}):(${validName})(:${validName})?(@${validName})?/)[0]
```

Comment:
I am not an expert on groovy and am still trying to learn the idiomatic way of doing things :)
But it looks like this makes codenarc happy, it seems to discourage `def` because it hides typing information:
``
  static List<String> artifactParts(String artifactCoordinate) {
    String name, extension, groupId, artifact, version, classifier
    (name, extension) = artifactCoordinate.tokenize("@")
    (groupId, artifact, version, classifier) = name.tokenize(":")
    return [groupId, artifact, version, classifier, extension]
  }
```

Btw you can run codenarc with `./gradlew check`. The codenarc complaint about colons can be resolved by removing the spaces after the colons like so: "group:groupId,"